### PR TITLE
Update in-place experiment writing to use the new WriterContract API in slicedimage 4.0.0

### PR DESCRIPTION
**Pull request by ttung**
Test plan: in-place experiment writing tests pass.

Note that travis won't succeed until https://github.com/spacetx/slicedimage/pull/109/ is approved and released.

### CODE REVIEW

**Comment by njmei** [comment]
I get an `ImportError` with the `tonytung-url` branch and `slicedimage==4.0.0`:

```
File "/home/nicholas.mei/miniconda3/envs/mfish_pipeline/lib/python3.7/site-packages/starfish/core/experiment/experiment.py", line 21, in <module>
   from slicedimage.urlpath import pathjoin
ModuleNotFoundError: No module named 'slicedimage.urlpath'
```

**Comment by ttung** [comment]
@njmei I made a mistake in an earlier refactor.  https://github.com/spacetx/slicedimage/pull/110/

File: starfish/core/experiment/builder/inplace.py [hunk 1]
```diff
@@ -1,64 +1,39 @@
 """To build experiments in-place, a few things must be done:
 
-1. Call enable_inplace_mode().
-2. Call write_experiment_json with tile_opener=inplace_tile_opener.
-3. The TileFetcher should return an instance of a InplaceTileFetcher.
-
-Please note that enabling in-place experiment construction should be only done in an isolated script
-dedicated to constructing an experiment, as it modifies some existing code paths.
+1. Call write_experiment_json with writer_contract=InplaceWriterContract.
```
Comment (njmei) [inline starfish/core/experiment/builder/inplace.py]: Should be:
```
writer_contract=InplaceWriterContract()
```

File: starfish/core/experiment/builder/__init__.py [hunk 2]
```diff
@@ -235,19 +227,28 @@ def write_irregular_experiment_json(
...
     if postprocess_func is None:
         postprocess_func = lambda doc: doc
-    if fov_path_generator is None:
-        fov_path_generator = _fov_path_generator
-    if tile_opener is None:
-        tile_opener = _tile_opener
+    if fov_path_generator is not None or tile_opener is not None:
+        warnings.warn(
+            "`fov_path_generator` and `tile_opener` options for writing experiment files is "
+            "deprecated.  Use `riter_contract` instead.",
```
Comment (shanaxel42) [inline starfish/core/experiment/builder/__init__.py]:
```suggestion
            "deprecated.  Use `writer_contract` instead.",
```

File: starfish/core/experiment/builder/inplace.py (hunk 1 above)
Comment (ttung) [reply starfish/core/experiment/builder/inplace.py]: Updated.

File: starfish/core/experiment/builder/__init__.py (hunk 2 above)
Comment (ttung) [reply starfish/core/experiment/builder/__init__.py]: Fixed!

**Pull request merged**